### PR TITLE
Add docs on distances and particles to the manual

### PR DESCRIPTION
**Pull request by markdewing**
Some development documentation on how to program ParticleSets, distance tables, and particle groups.

### CODE REVIEW

File: manual/distance_tables.tex [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+\subsection{ParticleSets}
+The \texttt{ParticleSet} class stores particle positions and attributes (charge, mass, etc).
+
+The \texttt{R} member stores positions.
+For calculations, the \texttt{R} variable needs to be transferred to the structure-of-arrays (SoA) storage in \texttt{RSoA}.   This is done by the \texttt{update} method.
+In the future the interface may change to use functions to set and retrieve positions
+so the SoA transformation of the particle data can happen automatically.
+
+A particular distance table is retrieved with \texttt{getDistTable}.
+The first table in the list is the table for like particles (electron-electron or ion-ion), and is always present.
```
Comment (ye-luo) [inline manual/distance_tables.tex]: I think this behavior has been removed. The first table only depends on the first call of addTable.

File: manual/distance_tables.tex [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+The flag, \texttt{IsGrouped}, indicates if the particles are grouped or not.
+The particles will not be grouped if the elements are not grouped together in the input file.
+This ordering is usually the responsibility of the converters.
+
+Code can be written to only handle the grouped case, but put an assert or failure check if the particles are not grouped.  Otherwise the code will give wrong answers and it can be time-consuming to debug.
+
+\subsection{Distance tables}
+
+Distance tables store distances between particles.
+There are symmetric (AA) tables for distance between like particles (electron-electron or ion-ion) and asymmetric (BA) tables for distance between unlike particles (ion-electron)
```
Comment (ye-luo) [inline manual/distance_tables.tex]: I think electron-ion is more accurate for SoA.

File: manual/distance_tables.tex [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+The \texttt{Distances} and \texttt{Displacements} members contain the data.
+Individual elements are accessed with with the \texttt{()} operator.
```
Comment (ye-luo) [inline manual/distance_tables.tex]: If I remember correctly, `[i][j]` works but `(i,j)` doesn't for SoA.

File: manual/distance_tables.tex (hunk 1 above)
Comment (markdewing) [reply manual/distance_tables.tex]: Yes, this has changed.   Removed that sentence.

File: manual/distance_tables.tex (hunk 2 above)
Comment (markdewing) [reply manual/distance_tables.tex]: Changed.

File: manual/distance_tables.tex (hunk 3 above)
Comment (markdewing) [reply manual/distance_tables.tex]: The () operator works for Distances, but not for Displacements.   The [] syntax works for both - the example code was updated to use that instead.

**Pull request merged**